File: nightgrass/src/main/java/com/goodfeel/nightgrass/repo/UserRepository.java
package com.goodfeel.nightgrass.repo;

import com.goodfeel.nightgrass.data.User;
import org.springframework.data.repository.reactive.ReactiveCrudRepository;
import reactor.core.publisher.Mono;

public interface UserRepository extends ReactiveCrudRepository<User, Long> {

    Mono<User> findByOauthId(String oauthId);
}

File: nightgrass/src/main/kotlin/com/goodfeel/nightgrass/repo/UserRepository.kt
package com.goodfeel.nightgrass.repo

import com.goodfeel.nightgrass.data.User
import org.springframework.data.repository.reactive.ReactiveCrudRepository
import reactor.core.publisher.Mono

interface UserRepository : ReactiveCrudRepository<User, Long> {
    fun findByOauthId(oauthId: String): Mono<User>
    fun findByGuestId(guestId: String): Mono<User>
    fun deleteByGuestId(guestId: String): Mono<Void>
}
